### DISCUSSION TITLE
从其他lua文件导入全局配置？

### DISCUSSION THREAD

**Discussion by kenchztk** [xmake-io/xmake#2744] [Q&A]
### Is your feature request related to a problem? Please describe.

我定义了一个平台相关的全局配置``platform.lua``希望用于其他项目复用:
``` lua
add_rules("mode.debug", "mode.release")
set_languages("c11", "cxx11")
set_symbols("debug")

if is_plat("windows") then
    set_optimize("none")
    set_runtimes("MDd")
else
    set_optimize("none")
    set_config("arch", "x86_64")
    set_runtimes("MD")
end
```
然后尝试在其他项目的``xmake.lua``中通过``includes("../platform.lua")``
但是从实验结果来看,这些设置没有预想中那样在目标工程的``xmake.lua``中生效并作用

### Describe the solution you'd like

请问我这个想法是否可行？或是有哪些步骤发生了错误？

**Comment by waruqi** [comment]
配置是按 tree 接口，从顶层往下继承的，根配置无法继承子 xmake.lua 里面的设置

方法1，这些配置放 root 顶层根描述域设置，所有子 xmake.lua 继承
方法2，搞成 custom rule, on_load 设置这些配置，includes 进来，然后通过 `add_rules("platform")` 对不同项目生效配置
方法3，用 function wrap 下后，外面调用下

```lua
function load_common_configs()
    add_rules("mode.debug", "mode.release")
    set_languages("c11", "cxx11")
    set_symbols("debug")

    if is_plat("windows") then
        set_optimize("none")
        set_runtimes("MDd")
    else
        set_optimize("none")
        set_config("arch", "x86_64")
       set_runtimes("MD")
    end
end
```

root xmake.lua

```lua
load_common_configs()
```
